src/transfer.ts

```diff
@@ -25,6 +25,45 @@ export const transferStatusSchema = z.nativeEnum(TransferStatus, {
   description: 'transferStatusSchema',
 })
 
+// These enums are first segmented between transfers in and transfers out
+// since they both allow a slightly different set of statuses.
+// Transfer in statuses are further segmented, since once a transfer in
+// progresses to sending the user funds, providers are expected
+// to return new metadata about the transfer. This segmentation allows
+// us to encode those semantics into Zod schemas.
+export enum TransferInStatusPreTx {
+  TransferStarted = TransferStatus.TransferStarted,
+  TransferFiatFundsDebited = TransferStatus.TransferFiatFundsDebited,
+  TransferReceivedFiatFunds = TransferStatus.TransferReceivedFiatFunds,
+  TransferFailed = TransferStatus.TransferFailed,
```

Comment:
n.b. `TransferFailed` for transfers in _sometimes_ requires a TX hash to be present, but not always, so it's in the `TransferInStatusPreTx` enum (e.g., transfers can fail before crypto is sent, or after crypto is sent, if the blockchain transfer fails for some reason).